### PR TITLE
Add --version flag 

### PR DESCRIPTION
**Pull request by aramase**
fixes https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/15

```
root@k8s-agentpool1-32372105-vmss000000:/etc/kubernetes/secrets-store-csi-providers/azure# ./provider-azure --version
{"version":"addversion1210","buildDate":"2019-12-10-22:24","minDriverVersion":"v0.0.7"}
```

### CODE REVIEW

File: version.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	pv := providerVersion{
+		Version:          BuildVersion,
+		Date:             BuildDate,
+		MinDriverVersion: minDriverVersion,
+	}
+
+	var res []byte
+	var err error
+	if res, err = json.Marshal(pv); err != nil {
+		log.Fatalf("[error]: %v", err)
```
Comment (ritazh) [inline version.go]: we should return this err and exit(1) in main

File: main.go [hunk 2]
```diff
@@ -34,6 +37,10 @@ func main() {
 
 	setupLogger()
 
+	if *version {
+		printVersionAndExit()
```
Comment (ritazh) [inline main.go]: currently we only log err, should we exit if err?

Comment (ritazh) [reply main.go]: nevermind you are using log.Fatalf. disregard

File: version.go (hunk 1 above)
Comment (ritazh) [reply version.go]: disregard as you are using Fatalf

File: version.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	fmt.Printf(string(res) + "\n")
+	os.Exit(0)
```
Comment (ritazh) [inline version.go]: nit: in general we should return err and check for error or success in main and have main do the exit

Comment (aramase) [reply version.go]: done!

**Pull request merged**